### ISSUE TITLE
Vsync makes fps jump from 90 to 45, no inbetween

### ISSUE THREAD

**Issue by jlartois** [ValveSoftware/openvr#1652]
After watching Alex Vlachos' talk on Advanced VR Rendering, I have a question about what framerates are achievable when in SteamVR > Settings, the "refresh rate" is set to 90Hz.

Say that you have an application that consistently takes 14ms to render out one frame (i.e. to render the left and right eye views for one frame). This is more than the 11.11..ms threshold to achieve 90fps. In fact, if Vsync is not considered, 14ms per frame equals about 71fps.

Does this mean that because of how vr::VRCompositor()->WaitGetPoses() works, the application consistently renders at 45fps?

I modified hellovr_opengl for a project, and am sadly running into this problem. I am consistently hitting 45fps instead of the 71fps (more or less) that I expect (I did some benchmarking and am fairly certain that 45fps is too slow).

In summary: if the application's **theoretical framerate** is even a fraction of a second **slower than 11.11..ms per frame** (i.e. 90fps), does this mean that the application's **actual framerate is immediately lowered to 45fps? No inbetween?**

That seems like a developer nightmare. Is there anything one can do to get closer to the theoretical framerate of 71fps? Please do not suggest altering the application to get under that 11.11ms limit (yes you are right, but it is not the point of this issue).

Some possible solutions that I tried:

- Can VSync be disabled? I read about IVRDriverDirectModeComponent being one way of achieving this, but I am not writing a driver, just an application.
- Can the refresh rate in SteamVR > Settings be altered? It seems fixed at 90Hz for the HTC Vive pro.

**Comment by Rectus** [comment]
Yes, it will limit it to 45Hz if it can't make the frame rate. Both SteamVR and current headset hardware are designed around being used with VSync, I doubt it would be a good idea to disable it even if it was possible.

AFAIK the Vive Pro only supports a 90Hz refresh rate. Some headsets like the Valve Index has multiple options, the minimum is 80Hz though.

**Comment by jlartois** [comment]
Thanks you so much, this clears up my confusion.
